symbols/overlay29.yml

```diff
@@ -387,13 +448,45 @@ overlay29:
       address:
         NA: 0x22EB178
         EU: 0x22EBB28
+        JP: 0x22EC7E0
       description: |-
         The user entity attempts to switch places with the target entity (i.e. by the effect of the Switcher Orb). 
         
         The function checks for the Suction Cups ability for both the user and the target, and for the Mold Breaker ability on the user.
         
         r0: pointer to user entity
         r1: pointer to target entity
+    - name: FloorLoop
+      address:
+        NA: 0x22EBD08
+        EU: 0x22EC6B8
+        JP: 0x22ED370
+      description: |-
+        The main function which updates the state of the floor. Called in a loop by InitializeDungeon while FloorIsOver returns false.
+        
+        The parameter is always equal to 1 and is used as parameter for another unknown function.
```

Comment:
This isn't true, the code in `InitializeDungeon` is:

```c
uVar8 = 1;
do {
    FloorLoop(uVar8);
    iVar17 = FloorIsOver();
    uVar8 = 0;
} while (iVar17 == 0);
```

So r0 here must be a flag for "is this the first loop?"